estate/models/real_estate.py

```diff
@@ -111,6 +111,12 @@ def _onchange_date_available(self):
                     "message": _("The date is in the past")
                 }
             }
+    
+    @api.ondelete(at_uninstall=False)
+    def _unlink_if_state_valid(self):
+        for record in self:
```

Comment:
Meaningful names are highly encouraged (not a mistake, but a good practice)
```suggestion
        for estate in self:
```